azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/MultiOracleScenarioRunner.kt

```diff
@@ -178,38 +155,104 @@ class MultiOracleScenarioRunner<
+
+        fun apply(iteration: SystemIteration) =
+            if (system is MutableSystem<*, *>) {
+                system.applyIteration(iteration)
+                system
+            } else {
+                systemDefinition = systemDefinition.copy(
+                    commands = systemDefinition.commands + iteration.commands,
+                    actionGenerators = iteration.actionGenerators,
+                )
+                systemFactory.create(systemDefinition)
+            }
+
+        fun destroy() {
+            if (system is MutableSystem<*, *>) {
+                system.destroy()
+            }
+        }
+    }
+
+    private fun generateScenario(resultBuilder: OracleResult.Builder<AF, CF, QF, AGF>): OracleScenario<AF, QF, AGF> {
+        val actionGenerators = testInstance.runTask(TaskType.CreateActionGenerators, scenario) { implementation ->
+            val systemFactory =
+                implementation.createSystemFactory() as? ActionGeneratingSystemFactory<AF, CF, QF, AGF, *, ActionGeneratingSystem<AF, CF>>
+                    ?: throw IllegalStateException("Trying to generate actions, but system factory does not create systems with action generation capability")
+            scenario.givenActionGenerations(systemFactory.actionGeneratorFactory).map {
+                it to ActionGeneratorType.Given
+            } + scenario.whenActionGenerations(systemFactory.actionGeneratorFactory).map {
+                it to ActionGeneratorType.When
+            }
+        }.result!!
+        val generateTaskResult = testInstance.runTask(TaskType.GenerateActions, scenario) { implementation ->
+            val systemFactory =
+                implementation.createSystemFactory() as? ActionGeneratingSystemFactory<AF, CF, QF, AGF, *, ActionGeneratingSystem<AF, CF>>
+                    ?: throw IllegalStateException("Trying to generate actions, but system factory does not create systems with action generation capability")
+            val declarations = scenario.declarations(systemFactory.actionFactory)
+            val commands = scenario.commands(systemFactory.actionFactory)
+            val systemCursor = ActionGeneratingSystemCursor(systemFactory, SystemDefinition(
+                declarations = declarations,
+                commands = commands
+            ))
+            try {
+                actionGenerators.fold(scenario) { baseScenario, (generators, type) ->
+                    val commandsFromPreviousIteration =
+                        if (baseScenario is GeneratedScenario) {
+                            baseScenario.givenActionCreators.map { it(systemFactory.actionFactory) } +
+                                baseScenario.whenActionCreators.map { it(systemFactory.actionFactory) }
+                        } else {
+                            emptyList()
+                        }
+                    val actionCreators = generateActions(systemCursor, commandsFromPreviousIteration, generators)
```

Comment:
❓ If I understand this correctly, the `commandsFromPreviousIteration` are not applied in this function, but rather in `generateActions`. If that is the case, when are the commands from the last iteration applied?